### DISCUSSION TITLE
How to show pure white texture in sceneform?

### DISCUSSION THREAD

**Discussion by udayk07** [sceneview/sceneform-android#219] [Q&A]
While trying to render a texture ( from Android View ), it showing as light grey instead of pure white. Used filmic tonemapper and tried other tone mappers ( like linear tone mapper ) also which does't work ( all cases texture renders as light grey ).

Tried to map inverseTonemap color to material's basecolor but that workaround also not works and also it makes red border into yellow color.

`material.baseColor.rgb = inverseTonemapSRGB(material.baseColor.rgb);`
![sceneform_AR](https://user-images.githubusercontent.com/78372917/144289530-672d8e9f-7063-4226-be6d-b99e7d0619d6.jpg)

Would anyone suggest steps to achieve pure white color texture in scene in sceneview ?

**Comment by ThomasGorisse** [comment]
I think you got 2 more ways to try:
- Completely disable the light estimation in case it was.
- Modify the texture emissive color instead of the base one.

**Reply by udayk07** [reply]
@ThomasGorisse Thanks for reply.
- Hope Light estimation only available in arSceneView. In this case, I used only sceneview.
- Tried Emissive color same as material base and few other colors ( black, white ). It seems not working as expected.

As of now, tried `toneMapper(new ToneMapper.Linear())` linear tonemapper which renders white color perfectly

**Reply by ThomasGorisse** [reply]
> As of now, tried `toneMapper(new ToneMapper.Linear())` linear tonemapper which renders white color perfectly

So you have your answer or I missed something?

We moved to Filmic Tone Mapper by default because of another use case where it was problematic. @RGregat should have the answer.

**Reply by udayk07** [reply]
Yeah,  as of now Linear tone mapper only solves this issue.

**Reply by RGregat** [reply]
We moved to Filmic because of washed out/pale colored textures when it comes to certain colors like red, orange, purple and so on. But that was before the lightning was updated by Thomas. Sure I can take a look, but Filmic looks still decent enough, maybe we should update the Document to address this and what alternatives exists. I guess one more thing for our ToDo @ThomasGorisse did I mentioned already that I have holiday from the 16 December until the end of the year :P I guess a good time to do some cleanup and documentation.

**Reply by udayk07** [reply]
@RGregat Filmic tone mapper looks good in visual than linear tone map. Only due to above issue ( not able to render pure white color ) moved to linear tone map. 

We render that white color box from ViewRenderable option. Is there any other way ( other linear tone map ) to render exact designed android view without changing color like simple view overlay to a model ( as text also seems little thin while render text ) ?

**Comment by ThomasGorisse** [comment]
@udayk07 I just had a quick thought about the fact that it could come from the default lights in case of a non AR usage.
Maybe the default values could be to bright since the AR estimations are applying ARCore factors to sunny day lights.

Could you try modifying the main and indirect light values from the SceneView to see if the issue really comes from the tone mapper?

**Reply by udayk07** [reply]
Sure. Will try with different light settings.

**Comment by ThomasGorisse** [comment]
Can you please install [AR Environment Lights](https://play.google.com/store/apps/details?id=com.gorisse.thomas.ar.environmentlights)?
And give the light estimations values you have.

**Comment by agGitHub** [comment]
I use Sceneform in a non-AR project and I have the same problem : I have a pure white plane but it doesn't seem white but gray. How to solve this problem ?

**Comment by ThomasGorisse** [comment]
Guys, I just have one question: In your house, if you have a blue light bulb, will your wall seem pure white?
Simply disable light estimation if you want not real colors.
If you want a pure white wall in your house without blue reflections, you will have to put a TV showing white image on it instead of white paint.
That's exactly the same for AR: If you want your plane to be pure white, you have to change its `emissiveColor` not its `baseColor`

On a non AR context, it will depend on the `sceneView.environment` which has a default hdr cubemap that could potentially be too dark and a `sceneView.mainLight` that might have too low intensity. But in any case, you will never get "pure white" since the lights might change the rendered `baseColor`
If you think those default values not correct, I will be happy to update them for more standard values. Just tell me what default values you want or PR it.

**Comment by agGitHub** [comment]
Thanks for your answer. Yes, I am perfectly aware of this behaviour but I think that the final effect is not what we would expect. Much better on iOS SceneKit. (ok, your Sceneform SDK is really great anyway :) )

In a non-AR context, how do you change the emissiveColor of a 3D object in Sceneform? Also, how do you change the 'sceneView.environment', the 'hdr cubemap', and the 'sceneView.mainLight' ? Is it possible to add several lights with different colors?

Thanks.
